### PR TITLE
feat(select): add experimental select

### LINKED ISSUE #1211: Experimental Component - Select - Development
**Development**
A developer can discover and install the experimental component by following accessible documentation.

- [ ] Add to carbon-components
- [ ] Update website (add to the experimental section and update component status page)

### PR DESCRIPTION
**Pull request by chrsfshr**
Closes #1211 

Adds in Experimental Select Component

Preview link: ....

### CODE REVIEW

File: src/components/select/_select.scss [hunk 1]
```diff
@@ -196,3 +197,204 @@
...
+
+    & ~ .#{$prefix}--label {
+      order: 1;
+    }
+
+    &:focus {
+      outline: none;
+      box-shadow: none;
+      border: 2px solid $brand-01;
+    }
```
Comment (alisonjoseph) [inline src/components/select/_select.scss:239]:
```
    &:focus {
      outline: 2px solid $focus;
      box-shadow: none;
    }
```

Using a border is causing a shift in width, would be good to swap to outline.

File: src/components/select/_select.scss [hunk 2]
```diff
@@ -196,3 +197,204 @@
...
+
+    &[data-invalid],
+    &[data-invalid]:focus {
+      // box-shadow: 0 2px 0 0 $support-01;
+      box-shadow: none;
+      border: 2px solid $support-01;
+    }
+
+    &:focus ~ .#{$prefix}--label {
+      color: $brand-01;
```
Comment (alisonjoseph) [inline src/components/select/_select.scss:253]:
```suggestion
      color: $focus;
```

**Comment by alisonjoseph** [comment]
Couple comments around focus state but otherwise LGTM!

File: src/components/select/_select.scss [hunk 3]
```diff
@@ -196,3 +197,204 @@
...
+    }
+
+    .#{$prefix}--select-optgroup {
+      color: $text-01;
+    }
+  }
+
+  .#{$prefix}--select--inline {
+    display: grid;
+    grid-template-columns: auto auto;
```
Comment (alisonjoseph) [inline src/components/select/_select.scss:319]: Does this work cross-browser? Probably better to use flexbox here

**Review by joshblack — CHANGES_REQUESTED** [review]
Looks great so far! Would echo @alisonjoseph's comments, and then also suggest we try and avoid the nested selectors under the `--select-input` selector.

**Comment by dakahn** [comment]
Hi. Testing on Chrome 69 on a Macbook using the aXe extension I found a few accessibility bugs:

- [ ] Looks like the color contrast on the Inline Select's label is too light. At 3.36:1 contrast for normal sized text it falls below WCAG AA spec. We're going to want closer to 4.50:1

- [ ] The color contrast for the category headings in the drop down itself is also far too light. Though I can't actually test the ratio on those menu items.

- [ ] Similar to the first point, Inline Select (Invalid) has the aforementioned color contrast label issues

- [ ] When the list box is in focus enter on the keyboard should expand the box.

For keyboard navigation reference [here's the relevant W3 page](https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html)

**Review by dakahn — CHANGES_REQUESTED** [review]
Testing on Chrome 69 on a Macbook using the aXe extension I found a few accessibility bugs:

- [ ] Looks like the color contrast on the Inline Select's label is too light. At 3.36:1 contrast for normal sized text it falls below WCAG AA spec. We're going to want closer to 4.50:1

- [ ] The color contrast for the category headings in the drop down itself is also far too light. Though I can't actually test the ratio on those menu items.

- [ ] Similar to the first point, Inline Select (Invalid) has the aforementioned color contrast label issues

- [ ] When the list box is in focus enter on the keyboard should expand the box.

For keyboard navigation reference [here's the relevant W3 page](https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html)

@aagonzales FYI the label text is too light for inline select.

**Comment by alisonjoseph** [comment]
@dakahn Can you check against a white background and see if the label contrast issue is still there. Unfortunately our dev env. isn't updated yet and still renders all of these on the gray.

**Comment by alisonjoseph** [comment]
@chrsfshr can you add a preview link and let Anna know when its up

**Comment by alisonjoseph** [comment]
nvm @dakahn it is on a white background. sorry

**Comment by dakahn** [comment]
@alisonjoseph No sweat! For the record I double checked any contrast errors flagged by aXe using the [WebAIM Contrast Checker.](https://webaim.org/resources/contrastchecker/) which gave me specific ratios against AA/AAA spec.

**Comment by alisonjoseph** [comment]
@dakahn Yep! thats what I use too, I tagged Anna above to take a look at updating it.

**Comment by alisonjoseph** [comment]
@aagonzales Looks like gray 50 is not accessbile on white 🤔 Suggest changing it to gray 60, #6f6f6f which passes. 

screen shot 2018-10-19 at 5 59 35 pm

**Comment by alisonjoseph** [comment]
Yea, the standard browser styles show up like this. Not sure how to handle the color contrast issue here. 
screen shot 2018-10-19 at 10 20 17 pm

**Pull request closed without merging**